### PR TITLE
NEOS-1641: Fixes stale DB Connections during Data Sync Workflow

### PR DESCRIPTION
**Pull request by nickzelei**
This is a pretty large refactor that does a few things:

- Consolidates DB Connection Caching into a single connection manager
- Updates SQL Manager to use ConnectionManager under the hood (part of the connection consolidation)
- Updates the Data Sync Workflow activities to properly release their connection sessions once they are finished with them
- Moves the connection reaper out of the Sync activity and into the startup script
- Centralizes the DB pools for the data sync activity to use a single pool that is shared across all (in-process) activities
- Updates the backend to use the same connection manager logic but to operate in the auto-close mode once a session is released since everything here is transactional
- Updates default connection timeout to be 10s and centralizes
- Updates a bunch of places that hardcoded postgres and pgx to use the module constants

### CODE REVIEW

File: backend/services/mgmt/v1alpha1/connection-data-service/connection-data_test.go [hunk 1]
```diff
@@ -1,956 +1,956 @@
...
-		},
-	})
-
-	require.Nil(t, err)
-	require.Len(t, resp.Msg.TableConstraints, 1)
-	require.EqualValues(t, map[string]*mgmtv1alpha1.UniqueConstraint{
-		"public.users": {Columns: []string{"id"}},
-	}, resp.Msg.TableConstraints)
-}
+// import (
```
Comment (alishakawaguchi) [inline backend/services/mgmt/v1alpha1/connection-data-service/connection-data_test.go:3]: did you mean to comment this out?

File: internal/connection-manager/manager.go [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+		groupConnMap:       map[string]map[string]T{},
+		config:             cfg,
+		shutdown:           make(chan any),
+		mu:                 &sync.Mutex{},
+	}
+}
+
+func (c *ConnectionManager[T]) GetConnection(
+	session SessionInterface,
+	connection ConnectionInput,
```
Comment (alishakawaguchi) [inline internal/connection-manager/manager.go:91]: might be helpful to have a few debug logs here about when it finds an existing vs when it creates a new one

File: worker/pkg/workflows/datasync/workflow/workflow_integration_test.go [hunk 3]
```diff
@@ -1573,31 +1573,32 @@ func (f *fakeEELicense) IsValid() bool {
 }
 
 func executeWorkflow(
-	t *testing.T,
+	t testing.TB,
```
Comment (alishakawaguchi) [inline worker/pkg/workflows/datasync/workflow/workflow_integration_test.go:1576]: 👍

**Review by alishakawaguchi — APPROVED** [review]
Very clean! Left a few comments nothing blocking

File: internal/connection-manager/manager.go (hunk 2 above)
Comment (nickzelei) [reply internal/connection-manager/manager.go:94]: Good callout. I updated the interface to require the logger and filled it out throughout with more messages.
This forced me to pass it down further anyways and replaced a few instances where I was using the global slog.

File: internal/testutil/testcontainers/redis/redis.go [hunk 4]
```diff
@@ -32,6 +34,10 @@ func (r *RedisTestContainer) Setup(ctx context.Context) (*RedisTestContainer, er
 		"docker.io/redis:7",
 		redis.WithSnapshotting(10, 1),
 		redis.WithLogLevel(redis.LogLevelVerbose),
+		testcontainers.WithWaitStrategy(
+			wait.ForLog("* Ready to accept connections"),
+			wait.ForExposedPort(),
```
Comment (nickzelei) [inline internal/testutil/testcontainers/redis/redis.go:39]: This seems to have fixed the redis container flakiness.

**Pull request merged**